libsolutil/Exceptions.h

```diff
@@ -54,6 +54,15 @@ struct Exception: virtual std::exception, virtual boost::exception
 		::boost::throw_line(__LINE__) \
 	)
 
+/// Throws an exception if condition is not met with a given description and extra information about the location the
+/// exception was thrown from.
+/// @param _condition if condition is not met, specified exception will be thrown.
+/// @param _exceptionType The type of the exception to throw (not an instance).
+/// @param _description The message that describes the error.
+#define solRequire(_condition, _exceptionType, _description) \
+	if (!(_condition))                                       \
+		solThrow(_exceptionType, _description)
```

Comment:
These two should also be in parentheses if possible:
```suggestion
		solThrow((_exceptionType), (_description))
```

Reply:
Ah ok, true. However, at least for the `_exceptionType` it is not possible. e.g. we have somewhere `solRequire(m_stackState == Empty, CompilerError, "");` that becomes eventually
```c++
if (!(m_stackState == Empty))
::boost::throw_exception(
	(CompilerError) () << ::solidity::util::errinfo_comment(((""))) << ::boost::throw_function("_function_name_")
					   << ::boost::throw_file("_file_name_") << ::boost::throw_line(431))
```

Where in this case the constructor of the exception is not called anymore.